### PR TITLE
Setup CI formatting check for Calyx-Py and Queues

### LINKED ISSUE #2251: Require that Python code conform to the Black formatter
Let's expand the scope of mandatory Black formatting in CI. [Currently](https://github.com/calyxir/calyx/blob/894525448d0269f044154d298c1216ea41d1e2c3/.github/workflows/format.yml#L42-L53) only two directories are tested, but I think this can safely be expanded to include all of `calyx-py` for instance.

Right now if I run `black calyx-py` I get
```
am3327@havarti:/scratch/anshuman/calyx$ black calyx-py/
...
All done! ✨ 🍰 ✨
22 files reformatted, 32 files left unchanged.
```

@polybeandip interested? Since it bit you once before and you figured this out in OCaml-land recently?

### PR DESCRIPTION
**Pull request by polybeandip**
Closes #2251 

Changes:
- Run `black --line-length 88` on `frontend/queues`, `calyx-py`, and `fud`
- Tweak `.github/workflows/format.yml` to check formatting for these directories
    - add the `--check` options to all calls to `black`

### CODE REVIEW

File: frontends/queues/badformat.py
Comment (anshumanmohan) [inline frontends/queues/badformat.py:1]: Will this get checked in or removed?

**Comment by polybeandip** [comment]
It fails when we have unformatted files now!!

**Review by anshumanmohan — APPROVED** [review]
A feast for the eyes!

File: frontends/queues/badformat.py
Comment (polybeandip) [reply frontends/queues/badformat.py:1]: removed it with [a296505](https://github.com/calyxir/calyx/pull/2295/commits/a2965053bb54d339b26280e2727521fa138f5dfe)

Comment (anshumanmohan) [reply frontends/queues/badformat.py:1]: Hehe I see now that you checked it in temporarily to watch CI fail. Perfect!

**Comment by rachitnigam** [comment]
Merging! Quick tip: marking PRs with "auto-merge" will allow them to be automatically merged when all the checks pass.

**Pull request merged**